swapfactory/utils.go

```diff
@@ -20,19 +21,35 @@ const (
 	StageCompleted
 )
 
+// StageToString converts a contract Stage enum value to a string
+func StageToString(stage byte) string {
+	switch stage {
```

Comment:
```suggestion
type Stage byte
const (
	StageInvalid Stage = iota //nolint:revive
	StagePending
	StageReady
	StageCompleted
)

// String converts a contract Stage enum value to a string
func (stage Stage) String() string {
	switch stage {
	// rest of code here
```

this will auto-format `Stage` into a string for printlns, logs, and such, may be overkill but could be handy later

Reply:
That is the way I originally wanted/tried to implement it. The problem was that the generated SwapFactory code is using `byte` for the stage. Explicitly calling a function seemed 10% more obvious than doing a cast to enable the String() method on the type.